src/sim/detail/InitializeTracks.cu

```diff
@@ -82,7 +82,7 @@ __global__ void init_tracks_kernel(const StatePointers            states,
             {
                 // Copy the geometry state from the parent for improved
                 // performance
-                TrackId::value_type parent_id
+                TrackId::size_type parent_id
```

Comment:
[minor suggestion] 

Even-though this is technically accurate (and typical of stl collection), it reads here a bit off.
```suggestion
                TrackId::id_type parent_id
                TrackId::index_type parent_id
```
This would have the added advantage of giving us the ability to have the `id_type` and the `size_type` being different (with the `id_type` carrying some action and/or additional information) without having to change the using code.

Reply:
Yeah, I struggled a bit with this one. I've used `index_type` before in other codes, and briefly in this one, but it's nonstandard and basically has the same meaning as `size_type` does. What do you mean by `id_type` and `size_type` being different? The OpaqueId has only two template properties: the size of the integer being used, and the class that's associated with it (in most cases, the type of  the object corresponding to the index in question).

Reply:
On further reflection I'm still leaning toward `size_type` being the right way to go (`size_type` is always an unsigned integer used for maximum capacity of a container -- and that's the meaning it has in the context of an opaque ID). We could also change to `index_type` or `int_type`, but those are less standard.

The `value_type` I'm less sure about because the `OpaqueId` doesn't actually use the value type -- it signifies the type of the "dereferenced" value.

Reply:
I still (weakly) prefer `id_type` which has no implication whatsoever on the backend implementation. (i.e. It sounded like we don't expect to use the OpaqueId or its content as an iterator so the reference to `size_type` is sorta moot).